File: python/b03.py
N = int(input())
A = list(map(int, input().split()))

isFound = False
for i in range(N):
    for j in range(i+1, N):
        for k in range(j+1, N):
            if A[i] + A[j] + A[k] == 1000:
                isFound = True
                break

if isFound:
    print("Yes")
else:
    print("No")

File: rust/b03.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        a: [u32; n],
    }
    let mut is_found: bool = false;
    for i in 0..n {
        for j in (i + 1)..n {
            for k in (j + 1)..n {
                if a[i] + a[j] + a[k] == 1000 {
                    is_found = true;
                }
            }
        }
    }
    if is_found {
        println!("Yes")
    } else {
        println!("No")
    }
}
